src/widgets.cpp

```diff
@@ -387,20 +388,19 @@ void Dropdown::draw() {
 			l.font = m.labelsize_ || m.labelfont_ ? m.labelfont_ : textfont();
 			l.size = m.labelsize_ ? m.labelsize_ : textsize();
 			l.color= m.labelcolor_ ? m.labelcolor_ : textcolor();
+			l.h_margin_ = l.v_margin_ = l.spacing = 0;
 			if (!m.active()) l.color = fl_inactive((Fl_Color)l.color);
 			fl_draw_shortcut = 2; // hack value to make '&' disappear
-			l.draw(xx+3, yy, ww, hh, FL_ALIGN_LEFT);
+			l.draw(xx+3, yy, ww>6 ? ww-6 : 0, hh, FL_ALIGN_LEFT);
 			fl_draw_shortcut = 0;
-			fl_pop_clip();
 			if (Fl::focus() == this) { draw_focus(box(), xx, yy, ww, hh); }
 		}
 		else {
-			fl_push_clip(xx, yy, ww, hh);
 			fl_draw_shortcut = 2; // hack value to make '&' disappear
 			m.draw(xx, yy, ww, hh, this, Fl::focus() == this);
 			fl_draw_shortcut = 0;
-			fl_pop_clip();
 		}
+		fl_pop_clip();
 	}

```

Comment:
The "origin" of this change is simply that the `Fl_Label` struct had a few new fields added to it (`h_margin_`, `v_margin_`, and `spacing`) which were all uninitialized in this code.

This resulted in the "Palettes" dropdown being drawn incorrectly/unpredictably.

Instead of "just" fixing the uninitialized data, I rewrote this based on the latest `Fl_Choice::draw()` while preserving most of the differences, since there are substantial useful changes, especially in the first half.

I couldn't fully decipher the purpose the custom changes to the second half (the displacement of `fl_push_clip()`/`fl_pop_clip()`, the recalculation of `ww` on old line 380) so I just went with the latest `Fl_Choice::draw()` and the final result looks the same to me AFAICT. If you have a reason for these customizations, let me know.

I originally did this in https://github.com/dannye/crystal-tracker/commit/fb01b5b653bbd1bbfbc9e092d90276e47f464abb